earth_enterprise/rpms/post-install_server.sh

```diff
@@ -0,0 +1,324 @@
+#!/bin/bash
+#
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: requires xmllint from libxml2-utils
+
+set +x
+
+#-----------------------------------------------------------------
+# from common.sh:
+
+BADHOSTNAMELIST=(empty linux localhost dhcp bootp)
+
+# versions and user names
+GEE="Google Earth Enterprise"
+GEES="$GEE Server"
+LONG_VERSION="5.2.1"
+
+MACHINE_OS=""
+
+# directory locations
+BININSTALLROOTDIR="/etc/init.d"
+BININSTALLPROFILEDIR="/etc/profile.d"
+BASEINSTALLLOGROTATEDIR="/etc/logrotate.d"
+BASEINSTALLDIR_OPT="/opt/google"
+BASEINSTALLDIR_ETC="/etc/opt/google"
+BASEINSTALLDIR_VAR="/var/opt/google"
+TMPINSTALLDIR="/tmp/fusion_os_install"
+INITSCRIPTUPDATE="/usr/sbin/update-rc.d"
+CHKCONFIG="/sbin/chkconfig"
+KH_SYSTEMRC="/usr/keyhole/etc/systemrc"
+
+# derived directories
+BASEINSTALLGDALSHAREDIR="$BASEINSTALLDIR_OPT/share/gdal"
+GENERAL_LOG="$BASEINSTALLDIR_VAR/log"
+INSTALL_LOG_DIR="$BASEINSTALLDIR_OPT/install"
+SYSTEMRC="$BASEINSTALLDIR_ETC/systemrc"
+FUSIONBININSTALL="$BININSTALLROOTDIR/gefusion"
+SOURCECODEDIR=$(dirname $(dirname $(readlink -f "$0")))
+TOPSOURCEDIR_EE=$(dirname $SOURCECODEDIR)
+GESERVERBININSTALL="$BININSTALLROOTDIR/geserver"
+GEHTTPD="$BASEINSTALLDIR_OPT/gehttpd"
+GEHTTPD_CONF="$GEHTTPD/conf.d"
+
+run_as_user() {
+    local use_su=`su $1 -c 'echo -n 1' 2> /dev/null  || echo -n 0`
+    if [ "$use_su" -eq 1 ] ; then
+        >&2 echo "cd / ;su $1 -c \"$2\""
+        ( cd / ;su $1 -c "$2" )
+    else
+        >&2 echo "cd / ;sudo -u $1 $2"
+        ( cd / ;sudo -u $1 $2 )
+    fi
+}
+
+#-----------------------------------------------------------------
+
+# config values
+PUBLISHER_ROOT="/gevol/published_dbs"
+DEFAULTGROUPNAME="gegroup"
+GROUPNAME=$DEFAULTGROUPNAME
+DEFAULTGEFUSIONUSER_NAME="gefusionuser"
+GEFUSIONUSER_NAME=$DEFAULTGEFUSIONUSER_NAME
+
+# user names
+GRPNAME="gegroup"
+GEPGUSER_NAME="gepguser"
+GEAPACHEUSER_NAME="geapacheuser"
+
+SERVER_INSTALL_OR_UPGRADE="install"
+GEE_CHECK_CONFIG_SCRIPT="/opt/google/gehttpd/cgi-bin/set_geecheck_config.py"
+PGSQL_DATA="/var/opt/google/pgsql/data"
+PGSQL_LOGS="/var/opt/google/pgsql/logs"
+PGSQL_PROGRAM="/opt/google/bin/pg_ctl"
+PRODUCT_NAME="$GEE"
+
+#-----------------------------------------------------------------
+# Main Functions
+#-----------------------------------------------------------------
+
+main_postinstall()
+{
+  # 1) Modify files
```

Comment:
Minor: We should decide on whether we want 2-space or 4-space indentation.

Reply:
ok.